src/services/LiveSubmissionsService.js

```diff
@@ -4,17 +4,24 @@ import VerdictIssuedEvent from "../models/VerdictIssuedEvent";
 
 export class LiveSubmissionsService extends AbstractService {
 
-    constructor(studentService, problemService, stompClient) {
+    constructor(studentService, problemService, submissionService, stompClient) {
         super({
             baseURL: process.env.REACT_APP_PROBLEM_SVC_BASE_URL,
             timeout: 10000,
             tokenSupplier: studentService.currentToken
         });
         this.studentService = studentService;
         this.problemService = problemService;
+        this.submissionService = submissionService;
         this.stompClient = stompClient;
     }
 
+    async queryExamLatestSubmissions(examId) {
+        let queryParameters = {examId, sortBy: 'submissionTime', ascending: false};
+        return await this.submissionService.getSubmissions(queryParameters)
+            .then(submissions => Promise.all(submissions.map(submission => this.completeLiveSubmissionFields(submission))));
+    }
```

Comment:
The nested `Promise.all` seems strange, would it possible to re-write it as `Promise.all(arr)` where you put all you need to wait into a single `arr`?